### ISSUE TITLE
Float32 for GPT-J-6B

### ISSUE THREAD

**Issue by Jacobsonradical** [baoguangsheng/fast-detect-gpt#25]
Hi, thanks again for the nice paper!

The best reference model is GPT-J-6B, and from the code I see you set it to Float16 precision. If I initialize the model with full Float32 precision, would it further improve the performance?

In addition, is there a solution for batch processing of texts instead of doing it sequentially to fully utilize the GPU?

Many thanks!

**Comment by baoguangsheng** [comment]
From what we've noticed, the float32 doesn't seem to enhance the performance. 

Please consider modifying the algorithm in the fast_detect_gpt.py to accommodate batch processing, as it currently presumes that logits.shape[0] is equal to 1. 

I hope this proves to be useful for you.

**Comment by Jacobsonradical** [comment]
Great. Thank you very much.
